core/domain/user_services.py

```diff
@@ -920,6 +940,63 @@ def has_fully_registered_account(user_id):
         feconf.REGISTRATION_PAGE_LAST_UPDATED_UTC)
 
 
+def get_all_profiles_auth_details_by_parent_user_id(parent_user_id):
+    """Gets domain objects representing the auth details for all profiles
+    associated with the user having given parent_user_id.
+
+    Args:
+        parent_user_id: str. User id of the parent_user whose associated
+            profiles we are querying for.
+
+    Returns:
+        list(UserAuthDetails). The UserAuthDetails domain objects
+        corresponding to the profiles linked to given parent_user_id. If that
+        parent user does not have any profiles linked to it, the
+        returned list will be empty.
+    """
+    return [
+        _get_user_auth_details_from_model(model) for model in
+        user_models.UserAuthDetailsModel.get_all_profiles_by_parent_user_id(
+            parent_user_id)
+    ]
+
+
+def is_display_alias_unique_within_the_account(parent_user_id, display_alias):
+    """Checks whether a given display alias is unique within the account.
+
+    Args:
+        parent_user_id: str. User id of the parent(full) user in the account.
+        display_alias: str. The queried display alias to be checked.
+
+    Returns:
+        bool. Whether the display alias is unique within the account among all
```

Comment:
You can make this clearer; see @BenHenning's comment above.

Reply:
Done.